### PR TITLE
`light-client`: Replace `serde_cbor` with `serde_json`

### LINKED ISSUE #1026: Use an alternative crate for `serde_cbor`
On the [project GitHub](https://github.com/pyfisch/cbor), it is indicated that the `serde_cbor` project has been archived since August 2021. The project suggests either [ciborium](https://crates.io/crates/ciborium) or [minicbor](https://crates.io/crates/minicbor) as alternatives.

### LINKED ISSUE #1038: RUSTSEC-2021-0127: serde_cbor is unmaintained
> serde_cbor is unmaintained

| Details             |                                                |
| ------------------- | ---------------------------------------------- |
| Status              | unmaintained                |
| Package             | `serde_cbor`                      |
| Version             | `0.11.2`                   |
| URL                 | [https://github.com/pyfisch/cbor](https://github.com/pyfisch/cbor) |
| Date                | 2021-08-15                         |

The `serde_cbor` crate is unmaintained. The author has archived the github repository.

Alternatives proposed by the author:

 *[`ciborium`](https://crates.io/crates/ciborium)
 *[`minicbor`](https://crates.io/crates/minicbor)

See [advisory page](https://rustsec.org/advisories/RUSTSEC-2021-0127.html) for additional details.

### PR DESCRIPTION
**Pull request by devashishdxt**
Fixes #1026.
Fixes #1038.

* [x] Referenced an issue explaining the need for the change
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Added entry in `.changelog/`

### CODE REVIEW

File: light-client/Cargo.toml [hunk 1]
```diff
@@ -47,12 +47,12 @@ derive_more = { version = "0.99.5", default-features = false, features = ["displ
 futures = { version = "0.3.4", default-features = false }
 futures-timer = { version = "3.0", default-features = false }
 serde = { version = "1.0.106", default-features = false }
-serde_cbor = { version = "0.11.1", default-features = false, features = ["alloc", "std"] }
 serde_derive = { version = "1.0.106", default-features = false }
 sled = { version = "0.34.3", optional = true, default-features = false }
 static_assertions = { version = "1.1.0", default-features = false }
 time = { version = "0.3.5", default-features = false, features = ["std"] }
 tokio = { version = "1.0", default-features = false, features = ["rt"], optional = true }
+serde_json = { version = "1.0.81", default-features = false, optional = true }
```
Comment (xla) [inline light-client/Cargo.toml:55]: Dependencies are order alphanumerically.

**Review by xla — CHANGES_REQUESTED** [review]
Bice, just a small nit.

File: light-client/Cargo.toml (hunk 1 above)
Comment (devashishdxt) [reply light-client/Cargo.toml]: Done

**Review by xla — APPROVED** [review]
٩(˘◡˘)۶

**Comment by devashishdxt** [comment]
@xla This PR was merged but I don't see these changes in `main` branch.

**Comment by thanethomson** [comment]
`main` is effectively a fork off of tendermint-rs v0.23.x, since Tendermint Core v0.35/v0.36 were abandoned. See https://github.com/informalsystems/tendermint-rs/discussions/1179

This would need to be ported across to `main` again.

**Pull request merged**